### DISCUSSION TITLE
Limits

### DISCUSSION THREAD

**Discussion by AL-THE-BOT-FATHER** [iSarabjitDhiman/TweeterPy#21] [General]
Have you tested the limits of the guest token?
How long do they last?
How often can you query?
Can you make multiple tokens?

Thank you for your amazing work.

**Comment by iSarabjitDhiman** [comment]
> Have you tested the limits of the guest token?
> How long do they last?
> How often can you query?
> Can you make multiple tokens?
> 
> Thank you for your amazing work.

1. Twitter made some major changes lately. Now it requires almost all the users to be logged in before they start surfing the website. But there are still some endpoints which are available to the guest users. But what I have observed is that these endpoints have some limits. For instance, while requesting for user tweets, it won't let u get past the first page (50~100 tweets), so no pagination is available to the guest users. There might be some other limitations as well, I haven't explored that much. 
2. Well, each endpoint has its own limits, where some of these take 15 minutes to cool down, others take a bit longer. You can check the api_limit_stats (async-await branch)
3. Even though there are limits but if you are getting just the basic data without any pagination, you can get as much data as you want as long as you respect the api limits. 
4. Yes you can, I mean the public token is going to be the same all the time. But you can generate the multiple guest sessions only if you are using some proxies/multiple IP addresses, otherwise you gonna have to wait until your api rate limit gets renewed. 

I hope I answered all of your queries, let me know if you still need any help.

**Comment by AL-THE-BOT-FATHER** [comment]
Thanks!
